examples/example-golang/main.go

```diff
@@ -68,6 +68,10 @@ func main() {
 		if !ok {
 			continue
 		}
+		if wf.Status.Phase == wfv1.NodeFailed || wf.Status.Phase == wfv1.NodeError {
```

Comment:
won't the block on line 75 also catch this?

Reply:
That's true. I was debugging why the example has been running forever and turns out that the problem is something else. I have updated this PR to print out the detailed error message from the status and added the timeout to the watch to improve the usability of this example. Let me know what you think.